tests/test_flow.py

```diff
@@ -2108,3 +2108,23 @@ def test_writer_downstream(tmpdir):
     controller.terminate()
     result = controller.await_termination()
     assert result == 45
+
+
+def test_complete_in_error_flow():
+    reduce = build_flow([
+        Complete(),
+        Reduce(0, lambda acc, x: acc + x)
+    ])
+    controller = build_flow([
+        Source(),
+        Map(lambda x: x + 1),
+        Map(RaiseEx(5).raise_ex, recovery_step=reduce),
+        reduce
```

Comment:
It's probably a good idea to make the recovery step different from the normal outlet, so that the test verifies that it actually went into recovery. E.g. add a `Map(lambda x: x+100)` to the recovery flow.